numpy/typing/tests/data/reveal/linalg.pyi

```diff
@@ -18,6 +18,7 @@ AR_c16: npt.NDArray[np.complex128]
 AR_O: npt.NDArray[np.object_]
 AR_m: npt.NDArray[np.timedelta64]
 AR_S: npt.NDArray[np.str_]
+AR_b: npt.NDArray[np.bool]
```

Comment:
Unrelated, but @BvB93, is there a way to make the error you get from `npt.NDArray[np.bool_]` nicer? I bet there are downstream type stubs that are broken now. In an older version of this PR this line triggered the following error:

```
E   AssertionError: Reveal mismatch at line 21
    
    error:
    Name "np.bool_" is not defined  [name-defined]
        AR_b: npt.NDArray[np.bool_]
        ^~~~~~~~~~~~~~~~~~~~~~~~~~
```


But both `np.bool` and `np.bool_` are defined right now in the main numpy namespace and neither are deprecated, so the typing should probably accept both or somehow warn that `bool_` is deprecated.

Reply:
> Unrelated, but @BvB93, is there a way to make the error you get from `npt.NDArray[np.bool_]` nicer?

Unfortunately not yet, no. There is [PEP 702: Marking deprecations using the type system](https://peps.python.org/pep-0702/) in the works which might help with this in the future, but even then both numpy and mypy will have to add some support for it before seeing the actual benefits.